pkg/mpc-examples/contracts/00-null-controller/NullControllerFactory.sol

```diff
@@ -0,0 +1,124 @@
+    }
+
+    event ControllerCreated(address indexed controller, bytes32 poolId);
+    event Disabled();
+
+    constructor(IVault vault, address factory) {
+        balancerVault = vault;
+        managedPoolFactory = factory;
+    }
+
+    /**
+     * @dev Return the address of the most recently created pool.
+     */
+    function getLastCreatedPool() external view returns (address) {
+        return _lastCreatedPool;
+    }
+
+    function create(MinimalPoolParams memory minimalParams) external {
+        require(!isDisabled, "Controller factory disabled");
+        require(!IManagedPoolFactory(managedPoolFactory).isDisabled(), "Pool factory disabled");
+
+        bytes32 controllerSalt = bytes32(_nextControllerSalt);
+        _nextControllerSalt += 1;
+
+        bytes memory controllerCreationCode = abi.encodePacked(
+            type(NullController).creationCode,
+            abi.encode(balancerVault)
+        );
+        address expectedControllerAddress = Create2.computeAddress(controllerSalt, keccak256(controllerCreationCode));
+
+        // build arguments to deploy pool from factory
+        address[] memory assetManagers = new address[](minimalParams.tokens.length);
+        for (uint256 i = 0; i < assetManagers.length; i++) {
+            assetManagers[i] = expectedControllerAddress;
+        }
+
+        // Populate IManagedPoolFactory.NewPoolParams with arguments from MinimalPoolParams and
+        // other arguments that this factory provides itself.
+        IManagedPoolFactory.NewPoolParams memory fullParams;
+        fullParams.name = minimalParams.name;
+        fullParams.symbol = minimalParams.symbol;
+        fullParams.tokens = minimalParams.tokens;
+        fullParams.normalizedWeights = minimalParams.normalizedWeights;
+        // Asset Managers set to the controller address, not known by deployer until creation
+        fullParams.assetManagers = assetManagers;
+        fullParams.swapFeePercentage = minimalParams.swapFeePercentage;
+        fullParams.swapEnabledOnStart = minimalParams.swapEnabledOnStart;
+        // Factory enforces public LPs for MPs with NullController
+        fullParams.mustAllowlistLPs = false;
+        fullParams.managementAumFeePercentage = minimalParams.managementAumFeePercentage;
+        fullParams.aumFeeId = minimalParams.aumFeeId;
+
+        IManagedPool pool = IManagedPool(
+            IManagedPoolFactory(managedPoolFactory).create(fullParams, expectedControllerAddress)
+        );
+        _lastCreatedPool = address(pool);
+
+        address actualControllerAddress = Create2.deploy(0, controllerSalt, controllerCreationCode);
+        require(expectedControllerAddress == actualControllerAddress, "Deploy failed");
+
+        // log controller locally
+        isControllerFromFactory[actualControllerAddress] = true;
+
+        // log controller publicly
+        emit ControllerCreated(actualControllerAddress, pool.getPoolId());
+    }
+
+    function disable() external onlyOwner {
```

Comment:
A short `@dev` comment here explaining what this function does and why it exists would probably be useful.